homeassistant/helpers/intent.py

```diff
@@ -398,13 +399,46 @@ class ServiceIntentHandler(IntentHandler):
     service_timeout: float = 0.2
 
     def __init__(
-        self, intent_type: str, domain: str, service: str, speech: str | None = None
+        self,
+        intent_type: str,
+        domain: str,
+        service: str,
+        speech: str | None = None,
+        extra_slot_names: set[str] | None = None,
```

Comment:
I don't think you need this. Just iterate `extra_slot_schema`, single source of truth.